src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -1069,7 +1076,10 @@ PyObject *c_parts_to_py (const rd_kafka_topic_partition_list_t *c_parts) {
 		PyList_SET_ITEM(parts, i,
 				TopicPartition_new0(
 					rktpar->topic, rktpar->partition,
-					rktpar->offset, rktpar->err));
+					rktpar->offset,
+					rktpar->metadata_size != 0 ?
```

Comment:
Since the c_parts has a shorter lifetime than this object the metadata needs to be copied (but preferably by new0).